### ISSUE TITLE
Tvhow seasons and episodes do not show up when using mysql backend

### ISSUE THREAD

**Issue by GoogleCodeExporter** [peteaky/android-xbmcremote#343]
```
What steps will reproduce the problem?
1.Go to market install xbmc-android
2.Connect to xbmc server using a mysql database backend
3. Browse through xbmc-android tvshows, pick a show and no seasons show up

What is the expected output? What do you see instead?
Should see seasons of the shows and then be able to move to episodes


What version of XBMC Remote are you using?
v0.8.3-beta1 Revision 686


Which Android device are you using (and which ROM, if custom)?
Verizon Droid


Which XBMC revision are you running and on which platform?
Linux XBMC PRE-10.5 r28256


Please provide any additional information below.

I have tracked this down through debugging on xbmc.

Output from log file using xbmc-android v0.8.3-beta1 below.


14:25:01 T:3078641536 M:3013410816   DEBUG: HttpApi Start command: 
QueryVideoDatabase  paras: QueryVideoDatabase; SELECT distinct(c12) FROM 
episode WHERE idEpisode IN  (SELECT idEpisode FROM tvshowlinkepisode WHERE 
idShow = 119 ) ORDER BY cast (c12 as integer)
14:25:01 T:3078641536 M:3013410816   DEBUG: CDatabase: Connecting to database 
xbmc_video at localhost:3306
14:25:01 T:3078641536 M:3013410816   ERROR: SQL: Undefined MySQL error: Code 
(1064)
                                            Query: SELECT distinct(c12) FROM episode WHERE idEpisode IN  (SELECT idEpisode FROM tvshowlinkepisode WHERE idShow = 119 ) ORDER BY cast (c12 as integer)
14:25:01 T:3078641536 M:3013410816   ERROR: GetArbitraryQuery failed



I have tracked this down to the code in 

XBMC Remote/src/org/xbmc/httpapi/client/TvShowClient.java

Attached is a patch to have queries that work with MySQL (mysql.patch)

I have also attached a patch that removes the "cast" from the queries, I have 
tested that it works with a mysql backed db but not an sqlitedb.

This seems to be the only two places in the httpapi.client code where queries 
are cast to an integer so i am not sure how important that is.


Hope this helps


```

Original issue reported on code.google.com by `konetzed@gmail.com` on 1 Aug 2010 at 7:30

Attachments:
- [mysql.patch](https://storage.googleapis.com/google-code-attachments/android-xbmcremote/issue-343/comment-0/mysql.patch)
- [all.patch](https://storage.googleapis.com/google-code-attachments/android-xbmcremote/issue-343/comment-0/all.patch)

**Comment by GoogleCodeExporter** [comment]
```
Cheers! Will test it as soon as possible. Thanks again!

Do you have it working with the other sections (movies and music) as well?
```

Original comment by `phree...@gmail.com` on 2 Aug 2010 at 6:53
- Changed state: **Accepted**
- Added labels: **Milestone-Release0.8**

**Comment by GoogleCodeExporter** [comment]
```
Set xbmc in debug mode and ran the remote through every screen and option I 
could think of and everything worked as expected for movies and music along 
with no errors in the log.  For testing this I used android-xbmcremote 
vv0.8.3-beta1. 


```

Original comment by `konetzed@gmail.com` on 3 Aug 2010 at 12:00

**Comment by GoogleCodeExporter** [comment]
```
konetzed, are you saying your solution works with mysql, but not with sqlite?
```

Original comment by `Gabor.Fo...@gmail.com` on 3 Aug 2010 at 4:13

**Comment by GoogleCodeExporter** [comment]
```
Gabor,  

There are two solutions I provided one that fixes the query for mysql that I 
know will break sqlite.  I provided it to just show the difference between the 
sqlite cast and the mysql cast.

The second solution I just removed the "cast" functions completely, to me the 
queries look syntactically correct for both mysql and sqlite.  Also looking at 
all the other queries I could not find another one that was casting an "order 
by" to an integer.  I have fully tested this solution for a mysql backed db but 
not sqlite.

So basically I cant find any reason why all.patch would not work for both mysql 
and sqlite.  I just have not tested it with sqlite as all my xbmc instances are 
now using a mysql backend.

If you guys know of a way of getting xbmc running in virtualbox I would be 
happy to setup a vm and test this functionality with a sqlite db. 

```

Original comment by `konetzed@gmail.com` on 3 Aug 2010 at 12:33

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Nice, i hope it work for me with last relase of ANDROID-XBMCREMOTE and 
xbmc-r32710-Dharma with MYSQL database.


THX.

```

Original comment by `garnier....@gmail.com` on 12 Aug 2010 at 12:11

**Comment by GoogleCodeExporter** [comment]
```
How can I implement these changes to my install?

I have used eclipse and SVN for a while, but cannot find where these lines 
should be put.
Is there a function in eclipse to auto-merge these patches?

Really like this function to be ready soon, so I can do some MySql testing at 
least.
```

Original comment by `zephy...@gmail.com` on 16 Aug 2010 at 12:54

**Comment by GoogleCodeExporter** [comment]
```
In eclipse you can use the Team - Apply Patch... in the context menu of the 
project if you have subclipse installed. However since the patch had been made 
the source changed, so you need to alter the source by hand I am afraid. In 
this case it is not so difficult because only two lines need to be changed. In 
the patch file you can see that the "XBMC 
Remote/src/org/xbmc/httpapi/client/TvShowClient.java" was corrected. Where the 
line starts with - , that needs to be deleted, where the line starts with + 
needs to be added.

konetzed said the "mysql.patch" does not work in sqlite, the "all.patch" 
removes the casting which is not a solution, since the database field is string 
type therefore without the cast the season 10 comes before season 2
```

Original comment by `Gabor.Fo...@gmail.com` on 16 Aug 2010 at 2:33

**Comment by GoogleCodeExporter** [comment]
```
Zephyr22

Gabor is correct the svn head has changed to much for my patches to apply 
cleanly ATM.  I am hoping to find a better solution and repost patches.

Gabor, do you have ideas on how to do this, I was thinking if there was maybe a 
way to set backend in the configuration and then have if statements to build 
the correct sql queries.  Or if there is some way through the api to find db 
type and build correct queries that way.  Not sure what is the best way please 
advise and I will work on fixes and submit patches.
```

Original comment by `konetzed@gmail.com` on 16 Aug 2010 at 4:24

**Comment by GoogleCodeExporter** [comment]
```
I am a small fish in this game. I think the owners of this project have more 
inside view into the xbmc. I have a strong feeling that because the mysql 
support is a new feature in the next xbmc version and the http api is an old 
code - actually taken over by the json rpc - it is not possible to detect the 
database type.
```

Original comment by `Gabor.Fo...@gmail.com` on 16 Aug 2010 at 6:22

**Comment by GoogleCodeExporter** [comment]
```
konetzed

The "ORDER BY c12+0" and "ORDER BY c12+0, c13+0" does the trick for both 
database engines.
```

Original comment by `Gabor.Fo...@gmail.com` on 16 Aug 2010 at 8:05

**Comment by GoogleCodeExporter** [comment]
```
Phreezie 

I have attached a patch in 302 that deals with issues introduced by the patch 
in r687 and also fixes these issues. 
```

Original comment by `konetzed@gmail.com` on 19 Aug 2010 at 11:48

**Comment by GoogleCodeExporter** [comment]
```
konetzed, can you confirm that this works now with the latest market version?
```

Original comment by `Gabor.Fo...@gmail.com` on 29 Jan 2011 at 10:29

**Comment by GoogleCodeExporter** [comment]
```
Confirmed working with latest version out of the market (v0.8.5-beta1) and xbmc 
10.0
```

Original comment by `konetzed@gmail.com` on 29 Jan 2011 at 5:17

**Comment by GoogleCodeExporter** [comment]
Original comment by `Gabor.Fo...@gmail.com` on 30 Jan 2011 at 3:10
- Changed state: **Fixed**
